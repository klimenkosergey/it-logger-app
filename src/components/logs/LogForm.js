import React, { Fragment, useState, useEffect } from 'react';

import TechSelectOptions from '../techs/TechSelectOptions';

const LogForm = props => {
  const [message, setMessage] = useState('');
  const [tech, setTech] = useState('');
  const [attention, setAttention] = useState(false);

  useEffect(() => {
    setMessage(props.message ? props.message : message);
    setTech(props.tech ? props.tech : tech);
    setAttention(props.attention ? props.attention : attention);
    // eslint-disable-next-line
  }, [props]);

  const onSubmit = e => {
    e.preventDefault();
    if (props.id) {
      props.onSubmit(props.id, message, tech, attention, new Date());
    } else {
      props.onSubmit(message, tech, attention, new Date());
    }

    // Clear inputs
    setMessage('');
    setTech('');
    setAttention(false);
  };

  return (
    <Fragment>
      <div className='modal-content'>
        <h4 className='center'>
          {!props.message && !props.tech ? 'Add' : 'Edit'} System Log
        </h4>
        <div className='row'>
          <div className='input-field'>
            <input
              type='text'
              name='message'
              value={message}
              onChange={e => setMessage(e.target.value)}
            />
            {!props.id && (
              <label htmlFor='message' className='active'>
                Log Message
              </label>
            )}
          </div>
        </div>
        <div className='row'>
          <div className='input-field'>
            <select
              name='tech'
              value={tech}
              className='browser-default'
              onChange={e => setTech(e.target.value)}
            >
              <option value='' disabled>
                Select Technician
              </option>
              <TechSelectOptions />
            </select>
          </div>
        </div>
        <div className='row'>
          <div className='input-field'>
            <label>
              <input
                type='checkbox'
                className='filled-in'
                checked={attention}
                value={attention}
                onChange={e => setAttention(!attention)}
              />
              <span>Needs Attention</span>
            </label>
          </div>
        </div>
      </div>
      <div className='modal-footer'>
        <a href='#!' onClick={onSubmit} className='waves-light blue btn'>
          Submit
        </a>
      </div>
    </Fragment>
  );
};

export default LogForm;
