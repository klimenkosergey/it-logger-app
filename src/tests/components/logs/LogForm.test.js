import React from 'react';
import { shallow, mount } from 'enzyme';

import LogForm from '../../../components/logs/LogForm';

let wrapper, onSubmit, log;

beforeEach(() => {
  log = {
    id: 1,
    message: 'Fixed hard drive',
    tech: 'Tony Stark'
  };
  onSubmit = jest.fn();
  wrapper = shallow(<LogForm onSubmit={onSubmit} {...log} />);
});

it('should render LogForm correctly', () => {
  wrapper = shallow(<LogForm onSubmit={onSubmit} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render LogForm correctly with provided log', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle onSubmit', () => {
  wrapper.find('.btn').simulate('click', { preventDefault: jest.fn() });
  expect(onSubmit).toHaveBeenCalled();
});

it('should change message on input change', () => {
  wrapper
    .find('input[name="message"]')
    .simulate('change', { target: { value: 'Server room' } });
  expect(wrapper.find('input[name="message"]').prop('value')).toBe(
    'Server room'
  );
});

it('should change tech on input change', () => {
  wrapper
    .find('select[name="tech"]')
    .simulate('change', { target: { value: 'Bruce Banner' } });
  expect(wrapper.find('select[name="tech"]').prop('value')).toBe(
    'Bruce Banner'
  );
});

it('should change attention on input change', () => {
  wrapper.find('input[type="checkbox"]').simulate('change');
  expect(wrapper.find('input[type="checkbox"]').prop('value')).toBe(true);
});
