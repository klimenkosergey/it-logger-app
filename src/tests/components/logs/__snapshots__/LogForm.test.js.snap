// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LogForm correctly 1`] = `
<Fragment>
  <div
    className="modal-content"
  >
    <h4
      className="center"
    >
      Add
       System Log
    </h4>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <input
          name="message"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="active"
          htmlFor="message"
        >
          Log Message
        </label>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <select
          className="browser-default"
          name="tech"
          onChange={[Function]}
          value=""
        >
          <option
            disabled={true}
            value=""
          >
            Select Technician
          </option>
          <Connect(TechSelectOptions) />
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <label>
          <input
            checked={false}
            className="filled-in"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
          <span>
            Needs Attention
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    className="modal-footer"
  >
    <a
      className="waves-light blue btn"
      href="#!"
      onClick={[Function]}
    >
      Submit
    </a>
  </div>
</Fragment>
`;

exports[`should render LogForm correctly with provided log 1`] = `
<Fragment>
  <div
    className="modal-content"
  >
    <h4
      className="center"
    >
      Edit
       System Log
    </h4>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <input
          name="message"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <select
          className="browser-default"
          name="tech"
          onChange={[Function]}
          value=""
        >
          <option
            disabled={true}
            value=""
          >
            Select Technician
          </option>
          <Connect(TechSelectOptions) />
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="input-field"
      >
        <label>
          <input
            checked={false}
            className="filled-in"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
          <span>
            Needs Attention
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    className="modal-footer"
  >
    <a
      className="waves-light blue btn"
      href="#!"
      onClick={[Function]}
    >
      Submit
    </a>
  </div>
</Fragment>
`;
